client/state/data-layer/wpcom/comments/index.js

```diff
@@ -156,6 +159,66 @@ export const announceFailure = ( { dispatch, getState }, { siteId, postId } ) =>
 	dispatch( errorNotice( error ) );
 };
 
+// @see https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/comments/%24comment_ID/delete/
+export const deleteComment = ( { dispatch, getState }, action ) => {
+	const { siteId, postId, commentId } = action;
+	const comment = getSiteComment( getState(), siteId, commentId );
+
+	dispatch( removeComment( siteId, postId, commentId ) );
```

Comment:
If we only have `COMMENTS_DELETE` we can update this to:

```jsx
dispatch( local( deleteComment( siteId, postId, commentId ) )
```
The `local` call, adds meta to the action to bypass the data layer, but can still be seen by the reducer.